src/System.Linq.Expressions/src/System/Linq/Expressions/UnaryExpression.cs

```diff
@@ -931,6 +931,15 @@ public static UnaryExpression Throw(Expression value, Type type)
         {
             ContractUtils.RequiresNotNull(type, nameof(type));
             TypeUtils.ValidateType(type, nameof(type));
+            if (type.IsByRef)
```

Comment:
these two checks happen together very often now. I wonder if it makes sense to combine them in a helper?